tests/unit/moduleapi/timer.tcl

```diff
@@ -54,7 +56,22 @@ start_server {tags {"modules"}} {
         assert_equal {} [r test.gettimer $id]
     }
 
-    test "Unload the module - timer" {
+    test "Module can be unloaded only when timer was finished" {
+        r set "timer-incr-key" 0
+        set id [r test.createtimer 2000 timer-incr-key]
+
+        # Make sure the Timer has not been fired
+        assert_equal 0 [r get timer-incr-key]
+        # Module can not be unload since the timer was ongoing 
+        assert_match {ERR*} [r module unload timer]
```

Comment:
```suggestion
    catch [r module unload timer] err
    assert_match {ERR*} $err
```

Reply:
you need to use catch, or you can use assert_error

Reply:
still not working, I think the error catch part was running parallel

Reply:
@warriorguo I tested it locally and it works, did you copy it wrongly(I edit it a few times)?

Reply:
you are right, it works! thank you!